src/main/scala/it/pagopa/interop/apigateway/server/impl/Dependencies.scala

```diff
@@ -27,18 +30,36 @@ import it.pagopa.interop.catalogmanagement.client.api.{EServiceApi => CatalogMan
 import it.pagopa.interop.commons.jwt.service.JWTReader
 import it.pagopa.interop.commons.jwt.service.impl.{DefaultJWTReader, getClaimsVerifier}
 import it.pagopa.interop.commons.jwt.{JWTConfiguration, KID, PublicKeysHolder, SerializedKey}
+import it.pagopa.interop.commons.logging.{CanLogContextFields, ContextFieldsToLog}
+import it.pagopa.interop.commons.ratelimiter.impl.RedisRateLimiter
 import it.pagopa.interop.commons.utils.TypeConversions.TryOps
+import it.pagopa.interop.commons.utils.errors.GenericComponentErrors
+import it.pagopa.interop.commons.utils.service.impl.OffsetDateTimeSupplierImpl
 import it.pagopa.interop.commons.utils.{AkkaUtils, OpenapiUtils}
 import it.pagopa.interop.notifier.client.api.EventsApi
 import it.pagopa.interop.purposemanagement.client.api.PurposeApi
 import it.pagopa.interop.selfcare.partymanagement.client.api.{PartyApi => PartyManagementApi}
+import it.pagopa.interop.tenantmanagement.client.api.{TenantApi => TenantManagementApi}
+import it.pagopa.interop.tenantprocess.client.api.{TenantApi => TenantProcessApi}
 
 import scala.concurrent.{ExecutionContext, ExecutionContextExecutor, Future}
-import it.pagopa.interop.tenantprocess.client.api.{TenantApi => TenantProcessApi}
-import it.pagopa.interop.tenantmanagement.client.api.{TenantApi => TenantManagementApi}
 
 trait Dependencies {
 
+  val rateLimiter: RateLimiter =
+    RedisRateLimiter(ApplicationConfiguration.rateLimiterConfigs, OffsetDateTimeSupplierImpl)
+
+  val rateLimiterDirective: ExecutionContext => Seq[(String, String)] => Directive1[Seq[(String, String)]] = {
+    ec => contexts =>
+      {
+        val logger: LoggerTakingImplicit[ContextFieldsToLog] = Logger.takingImplicit[ContextFieldsToLog](this.getClass)
+        RateLimiterDirective.rateLimiterDirective(
+          rateLimiter,
+          problemOf(StatusCodes.TooManyRequests, GenericComponentErrors.TooManyRequests)
+        )(contexts)(ec, entityMarshallerProblem, logger)
+      }
```

Comment:
why not in this way? is it possible?
```suggestion
        RateLimiterDirective.rateLimiterDirective(
          rateLimiter,
          problemOf(StatusCodes.TooManyRequests, GenericComponentErrors.TooManyRequests)
        )(contexts)(ec, entityMarshallerProblem, Logger.takingImplicit[ContextFieldsToLog](this.getClass))
```